assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js

```diff
@@ -66,36 +66,69 @@ export default function Idea( props ) {
 		dismissIdea,
 		setActivity,
 		removeActivity,
+		removeIdeaFromNewIdeas,
 		removeIdeaFromNewAndSavedIdeas,
+		moveIdeaFromSavedIdeasToNewIdeas,
+		moveIdeaFromNewIdeasToSavedIdeas,
 	} = useDispatch( MODULES_IDEA_HUB );
 
 	const activity = useSelect( ( select ) =>
 		select( MODULES_IDEA_HUB ).getActivity( name )
 	);
 
 	const handleDelete = useCallback( async () => {
-		setActivity( name, IDEA_HUB_ACTIVITY_IS_PROCESSING );
+		setActivity( name, IDEA_HUB_ACTIVITY_IS_DELETING );
 		await dismissIdea( name );
-		removeActivity( name );
+		setActivity( name, IDEA_HUB_ACTIVITY_DELETED );
 
 		trackEvent( IDEA_HUB_GA_CATEGORY_WIDGET, 'dismiss_idea' );
-	}, [ name, dismissIdea, setActivity, removeActivity ] );
+
+		await waitForActivity();
+		removeActivity( name );
```

Comment:
Probably this is also can be moved into the actions as well.

Reply:
My general thought is that it's preferable to keep delays like this in the component as it's more of a presentational concern rather than waiting for data. Moving it into the store would require creating a control for it as well.